### PR TITLE
Very basic umask implementation

### PR DESCRIPTION
**Pull request by benofbrown**
Doesn't yet support the -p or -S flags provided by bash but it does
allow for getting and setting the umask.

### CODE REVIEW

File: builtin/umask.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+int builtin_umask(struct mrsh_state *state, int argc, char *argv[]) {
+	mode_t mode;
+	mode_t default_mode = 0022;
+	char *endptr;
+	
+	if (argc < 2) {
+		mode = umask(default_mode);
+		printf("%04o\n", mode);
+		umask(mode);
+    return 0;
```
Comment (emersion) [inline builtin/umask.c]: Style: we use tabs

File: builtin/umask.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#include <sys/types.h>
+#include <sys/stat.h>
+#include "builtin.h"
+
+static const char umask_usage[] = "usage: umask [mode]\n";
+
+int builtin_umask(struct mrsh_state *state, int argc, char *argv[]) {
+	mode_t mode;
+	mode_t default_mode = 0022;
+	char *endptr;
```
Comment (emersion) [inline builtin/umask.c]: Can we declare this right before it's used?

File: builtin/umask.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	mode = strtol(argv[1], &endptr, 8);
+	if (*endptr != '\0') {
+		fprintf(stderr, umask_usage);
+		return 1;
+	}
+
+	umask(mode);
+	return 0;
```
Comment (emersion) [inline builtin/umask.c]: `return EXIT_SUCCESS` (also applies to other `return`s)

**Comment by benofbrown** [comment]
Thanks for the feedback, I'm at work now but I should be able to address the points you've raised this evening.

**Comment by emersion** [comment]
No worries, take your time!

**Comment by ddevault** [comment]
>Doesn't yet support the -p or -S flags provided by bash

Note that only the -S flag is specified by POSIX, so support for the -p flag is not suitable for inclusion in mrsh.

http://pubs.opengroup.org/onlinepubs/9699919799/utilities/umask.html

**Comment by emersion** [comment]
Thanks!

**Pull request merged**